pkg/nats/stream/types.go

```diff
@@ -63,27 +72,18 @@ type StreamConsumerClientConfig struct {
 	StreamCancellationBufferDuration time.Duration
 }
 
-// HeartBeatConfig represents the configuration of producer heart beating to the consumer.
-type HeartBeatConfig struct {
-	// HeartBeatIntervalDuration represents the duration between two heart beats from the producer client
-	// to consumer client.
-	HeartBeatIntervalDuration time.Duration
-	// HeartBeatRequestTimeout represents the time within which the producer client should receive the
-	// response from the consumer client.
-	HeartBeatRequestTimeout time.Duration
-}
-
 // HeartBeatRequest sent by producer client to the consumer client.
 type HeartBeatRequest struct {
-	ProducerConnID  string
-	ActiveStreamIds []string
+	// ActiveStreamIds is a map of active stream ids on producer client, where key is the RequestSubject, where
+	// the original request to initiate a streaming connection was sent.
+	ActiveStreamIds map[string][]string
 }
 
 // ConsumerHeartBeatResponse represents a heart beat response from the consumer client.
 type ConsumerHeartBeatResponse struct {
 	// NonActiveStreamIds represent a list of stream ids which the consumer client is not
 	// interested in.
```

Comment:
nit: please update comment and mention the key is the subject name where the value is the stream ID